lib/charms/zookeeper/v0/zookeeper_provider.py

```diff
@@ -0,0 +1,287 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+from collections import defaultdict
+import logging
+from typing import Dict, List, Optional, Set
+from kazoo.handlers.threading import KazooTimeoutError
+from kazoo.security import ACL, make_acl
+from ops.charm import RelationBrokenEvent, RelationEvent
+
+from ops.framework import Object
+from ops.model import MaintenanceStatus, Relation
+
+from charms.zookeeper.v0.client import (
+    MemberNotReadyError,
+    MembersSyncingError,
+    QuorumLeaderNotFoundError,
+    ZooKeeperManager,
+)
+from charms.zookeeper.v0.cluster import UnitNotFoundError, ZooKeeperCluster
+
+logger = logging.getLogger(__name__)
+
+REL_NAME = "database"
```

Comment:
The name of the relation on metadata.yaml is different.
```suggestion
REL_NAME = "zookeeper"
```